hedera-node/docs/system-accounts-operations.md

```diff
@@ -0,0 +1,146 @@
+
+**Hedera:**
+1. For addresses in range 0x0 - 0x2EE (0.0.0 - 0.0.750): returns 0 as if account doesn’t exist.
+2. For addresses in range 0x2EF - 0x3E8 (0.0.751 - 0.0.1000): returns balance or 0 if an account or contract doesn’t exist on the address. This behavior is the same for addresses above 0.0.1000.
+
+| Address                            | in Ethereum                                                                   | in Hedera                                                                     |
+|------------------------------------|-------------------------------------------------------------------------------|-------------------------------------------------------------------------------|
+| 0x0                                | ✅ returns balance or 0 if an account or contract doesn’t exist on the address | ✅ returns 0 as if account doesn’t exist                                       |
+| 0x1 → 0x9                          | ✅ returns balance or 0 if an account or contract doesn’t exist on the address | ✅ returns 0 as if account doesn’t exist                                       |
+| 0xA → 0x166                        | ✅ returns balance or 0 if an account or contract doesn’t exist on the address | ✅ returns 0 as if account doesn’t exist                                       |
+| 0x167 (0.0.359)                    | ✅ returns balance or 0 if an account or contract doesn’t exist on the address | ✅ returns 0 as if account doesn’t exist                                       |
+| 0x168, 0x169 (0.0.360, 0.0.361)    | ✅ returns balance or 0 if an account or contract doesn’t exist on the address | ✅ returns 0 as if account doesn’t exist                                       |
+| 0x16a → 0x2EE (0.0.361 → 0.0.750)  | ✅ returns balance or 0 if an account or contract doesn’t exist on the address | ✅ returns 0 as if account doesn’t exist                                       |
+| 0x2EF → 0x3E8 (0.0.751 → 0.0.1000) | ✅ returns balance or 0 if an account or contract doesn’t exist on the address | ✅ returns balance or 0 if an account or contract doesn’t exist on the address |
+| 0x3E8 → ♾️                         | ✅ returns balance or 0 if an account or contract doesn’t exist on the address | ✅ returns balance or 0 if an account or contract doesn’t exist on the address |
+
+### SelfDestruct operation
+
+**Ethereum:** success for any address. If the beneficiary is the same as contract address, it burns the eth and destructs the contract.
+
+**Hedera:**
+1. For addresses in range 0x0 - 0x2EE (0.0.0 - 0.0.750): fail with status `INVALID_SOLIDITY_ADDRESS`.
+2. For addresses in range 0x2EF - 0x3E8 (0.0.751 - 0.0.1000):
+   - success, if account exists and has `receiverSigRequired` == false;
+   - success, if account exists and has `receiverSigRequired` == true and account is `sender`;
+   - fail in some failing scenarios such as: beneficiary same as account-to-be-deleted, `receiverSigRequired` == true and signature provided, etc.
```

Comment:
We can add this information to the doc:
If the account to be deleted is the same as the beneficiary address, the operation would fail with status `SELF_DESTRUCT_TO_SELF`.
If the contract to be deleted is a token treasury, the operation would fail with status `CONTRACT_IS_TREASURY`.
If the contract to be deleted has a positive balance, the operation would fail with status `TRANSACTION_REQUIRES_ZERO_TOKEN_BALANCES`.
If the contract to be deleted owns NFTs, the operation would fail with status `CONTRACT_STILL_OWNS_NFTS`.
If the beneficiary has configured receiversigrequired = true on their account and are not the sender in the message frame, the operation would fail with status `INVALID_SIGNATURE`.

Reply:
Comment addressed.